framework/src/ppl/delite/framework/ops/DeliteOpsBaseGenericGen.scala

```diff
@@ -945,10 +948,48 @@ trait GenericGenDeliteOps extends BaseGenLoopsFat with BaseGenStaticData with Ba
       emitReturn("__act")
     }
 
+    val isTargetScala = (fileExtension == "scala")
+    val isTargetCpp = (fileExtension == "cpp")
+
+    def emitTimerStart(name: String) = {
+      if (isTargetScala) {
```

Comment:
You should never try to figure out the codegen target in the generic trait, but rather override / implement methods in DeliteOpsScalaGen and DeliteOpsCGen to provide target-specific strings where needed